Source/Syrup/Player/SyrupPlayerCharacter.cpp

```diff
@@ -18,12 +21,16 @@ ASyrupPlayerCharacter::ASyrupPlayerCharacter()
  	//Set this character to call Tick() every frame. You can turn this off to improve performance if you don't need it.
 	PrimaryActorTick.bCanEverTick = true;
 
+	//Define Resources
+	Energy = 10;
+	PlantSelected = 0;
+	PlantTypes = { NULL };
```

Comment:
You do not what the plant types array to ever contain null because the game would crash if the player tried to plant null. Should be initialized as empty.

Also nowhere in the documentation does it say you can initialize a TArray with an initializer list (the curly brackets), so even though this compiles and works I'd be suspicious of it if the array starts acting funky. I would recommend just initializing it as `= TArray<TSubclassOf<APlant>>();` since it should be empty anyways.